src/main/java/org/numenta/nupic/network/SerialConfig.java

```diff
@@ -0,0 +1,373 @@
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+package org.numenta.nupic.network;
+
+import java.nio.file.OpenOption;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+
+import org.numenta.nupic.ComputeCycle;
+import org.numenta.nupic.FieldMetaType;
+import org.numenta.nupic.Parameters;
+import org.numenta.nupic.Persistable;
+import org.numenta.nupic.algorithms.BitHistory;
+import org.numenta.nupic.algorithms.Classification;
+import org.numenta.nupic.model.Cell;
+import org.numenta.nupic.model.Column;
+import org.numenta.nupic.model.DistalDendrite;
+import org.numenta.nupic.model.ProximalDendrite;
+import org.numenta.nupic.model.Segment;
+import org.numenta.nupic.model.Synapse;
+import org.numenta.nupic.network.NetworkSerializer.Scheme;
+import org.numenta.nupic.util.NamedTuple;
+import org.numenta.nupic.util.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * <p>
+ * Used to configure a {@link NetworkSerializer} with details regarding file formats, 
+ * save locations, file opening conventions, and underlying serialization library,
+ * among other things.
+ * </p><p>
+ * <b>This file however provides defaults which may confidently be used for full featured
+ * use</b>, however more experienced users may want to tweak details (the first thing 
+ * being whether checkpoints add new files or delete previous files maintaining only 
+ * one file). For explanation of this see: {@link StandardOpenOptions} and the
+ * {@link #SerialConfig(String, Scheme, List, StandardOpenOption...)} constructor
+ * </p>
+ * <p>
+ * The check point file format is highly configurable using this class. The default format
+ * contains a "name" portion and a "timestamp" portion which may be independently configured.
+ * see {@link #setCheckPointFileName(String)} and {@link #setCheckPointTimeFormatString(String)}.
+ * </p>
+ * <p>
+ * In addition, you may also call {@link #setOneCheckPointOnly(boolean)} to overwrite the 
+ * checkpoint file if you would rather not maintain multiple checkpoints.
+ * 
+ * 
+ * @see NetworkTest
+ * @see JavaFstNetworkSerializationTest
+ * @see JavaKryoNetworkSerializationTest
+ * @see NetworkSerializer
+ * @see NetworkSerializerImpl
+ * 
+ * @author cogmission
+ */
+public class SerialConfig {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(NetworkSerializerImpl.class);
+
+    public static final Class[] DEFAULT_REGISTERED_TYPES = new Class[] {
+        Region.class, Layer.class, Cell.class, Column.class, Synapse.class,
+                ProximalDendrite.class, DistalDendrite.class, Segment.class, Inference.class,
+                ManualInput.class, BitHistory.class, Tuple.class, NamedTuple.class, Parameters.class,
+                ComputeCycle.class, Classification.class, FieldMetaType.class, Persistable.class
```

Comment:
That's fine I also added Pool...I will have to add Classification back in after the build runs